src/main/kotlin/org/jetbrains/research/testspark/settings/SettingsApplicationState.kt

```diff
@@ -39,7 +39,8 @@ data class SettingsApplicationState(
     var methodPrompt: String = DefaultSettingsApplicationState.methodPrompt,
     var linePrompt: String = DefaultSettingsApplicationState.linePrompt,
     var defaultLLMRequests: String = DefaultSettingsApplicationState.defaultLLMRequests,
-    var junitVersion: JUnitVersion = DefaultSettingsApplicationState.junitVersion,
+    var junitVersion: String = DefaultSettingsApplicationState.junitVersion,
```

Comment:
Why changing type `JUnitVersion` -> `String` ? It feels like it creates extra readability issues, overhead when extracting meta-info about junit

Reply:
unfortunatlly, `junitVersion` is not updated in the`settingsState` after a change. You can check it in the previous commits, it can't save complicated enum class for some reason (I've checked the xml files)

Reply:
It can be done by using [Converter class](https://plugins.jetbrains.com/docs/intellij/persisting-state-of-components.html#implementing-the-state-class) and because we have a kotlinx serialization dependency (need only to apply plugin), you can do it simply like this:
```
class JUnitVersionConverter : Converter<JUnitVersion>() {
    override fun fromString(value: String): JUnitVersion {
        return Json.decodeFromString(value)
    }

    override fun toString(value: JUnitVersion): String {
        return Json.encodeToString(value)
    }
}
```

and adding:

```
@OptionTag(converter = JUnitVersionConverter::class) var junitVersion ....
```

I did it myself and found out that `SettingsApplicationService` is not used as a project service. After fixing it, state saving seems to be working as intended. There are a lot of changes which need refactoring, and I'll send you a patch